### PR TITLE
Add List Property and Delete Property Command

### PR DESCRIPTION
**Pull request by jchilling**
- Refactor List -> ListBuyer
- Refactor Delete -> DeleteBuyer
- Add ListProperty Command
- Add DeleteProperty Command

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/CliSyntax.java [hunk 1]
```diff
@@ -16,4 +16,6 @@ public class CliSyntax {
     public static final Prefix PREFIX_PROPERTIES = new Prefix("-pr");
     public static final Prefix PREFIX_PRICE = new Prefix("-price"); // temporary, should refactor all prefixes
     public static final Prefix PREFIX_DESCRIPTION = new Prefix("-d");
+    public static final Prefix PREFIX_PROPERTY = new Prefix("-prop");
+    public static final Prefix PREFIX_BUYER = new Prefix("-buyer");
```
Comment (riccqi) [inline src/main/java/seedu/address/logic/parser/CliSyntax.java:20]: Where are these prefixes used? Do our commands actually need this?

File: src/test/java/seedu/address/logic/commands/ListPropertiesCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+public class ListPropertiesCommandTest {
+
+    private Model model;
+    private Model expectedModel;
+
+    @BeforeEach
+    public void setUp() {
+        model = new ModelManager(getTypicalPropertyBook(), new UserPrefs());
+        expectedModel = new ModelManager(model.getAddressBook(), new UserPrefs());
```
Comment (riccqi) [inline src/test/java/seedu/address/logic/commands/ListPropertiesCommandTest.java:23]: Will need to getPropertyBook once merged with zsig's PR

File: src/test/java/seedu/address/model/property/PropertyNameTest.java [hunk 3]
```diff
@@ -6,38 +6,35 @@
...
-        assertTrue(Name.isValidName("peter jack")); // alphabets only
-        assertTrue(Name.isValidName("12345")); // numbers only
-        assertTrue(Name.isValidName("peter the 2nd")); // alphanumeric characters
-        assertTrue(Name.isValidName("Capital Tan")); // with capital letters
-        assertTrue(Name.isValidName("David Roger Jackson Ray Jr 2nd")); // long names
+        assertTrue(PropertyName.isValidPropertyName("peter jack")); // alphabets only
+        assertTrue(PropertyName.isValidPropertyName("12345")); // numbers only
+        assertTrue(PropertyName.isValidPropertyName("peter the 2nd")); // alphanumeric characters
+        assertTrue(PropertyName.isValidPropertyName("Capital Tan")); // with capital letters
+        assertTrue(PropertyName.isValidPropertyName("David Roger Jackson Ray Jr 2nd")); // long names
```
Comment (riccqi) [inline src/test/java/seedu/address/model/property/PropertyNameTest.java:38]: nit: can change the sample names to something we're more likely to see for properties? eg Heng Mui Keng Street 2 5 Condo

File: src/test/java/seedu/address/testutil/TypicalProperties.java [hunk 4]
```diff
@@ -27,4 +32,20 @@ public class TypicalProperties {
...
+    /**
+     * Returns an {@code AddressBook} with all the typical persons.
+     */
+    public static AddressBook getTypicalPropertyBook() {
+        AddressBook ab = new AddressBook();
+        for (Property property : getTypicalProperties()) {
+            ab.addProperty(property);
+        }
+        return ab;
+    }
```
Comment (riccqi) [inline src/test/java/seedu/address/testutil/TypicalProperties.java:45]: Will need to use PropertyBook after @zsiggg's PR

**Review by riccqi — APPROVED** [review]
Overall LGTM! just some nits and questions

File: src/main/java/seedu/address/logic/parser/CliSyntax.java (hunk 1 above)
Comment (jchilling) [reply src/main/java/seedu/address/logic/parser/CliSyntax.java:20]: No, I thought about whether to make the command behave such that list -prop list the property list and list -buyer list the buyer list. But I decided to make two commands in the end. Will delete!

File: src/test/java/seedu/address/model/property/PropertyNameTest.java (hunk 3 above)
Comment (jchilling) [reply src/test/java/seedu/address/model/property/PropertyNameTest.java:38]: Ok!

**Pull request merged**